frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx

```diff
@@ -29,63 +30,47 @@ export default class TextPicker extends Component<*, Props, *> {
         placeholder: "Enter desired text"
     }
 
-    addValue() {
-        let values = this.props.values.slice();
-        values.push(null);
-        this.props.onValuesChange(values);
-    }
-
-    removeValue(index: number) {
-        let values = this.props.values.slice();
-        values.splice(index, 1);
-        this.props.onValuesChange(values);
-    }
-
-    setValue(index: number, value: string|null) {
-        let values = this.props.values.slice();
-        values[index] = value;
-        this.props.onValuesChange(values);
+    setValue(fieldString: ?string) {
+        if (fieldString != null && fieldString !== "") {
+            const newLineRegex = /\r?\n|\r/g;
+            const newValues = fieldString.replace(newLineRegex,'').split(',').map((value: string|null) => value);
```

Comment:
Probably want to do `.split(/,\s*/g)` so `foo, bar` doesn't become `["foo", " bar"]` (note the space before "bar")